packages/dids/src/methods/did-dht.ts

```diff
@@ -0,0 +1,1568 @@
+  }
+
+  /**
+   * Publishes a DID document to the DHT network.
+   *
+   * @param params - The parameters to use when publishing the DID document to the DHT network.
+   * @param params.did - The DID object whose DID document will be published.
+   * @param params.relay - The Pkarr relay to use when publishing the DID document.
+   * @returns A promise that resolves to `true` if the DID document was published successfully.
+   *          If publishing fails, `false` is returned.
+   */
+  public static async put({ did, relay }: {
+    did: Did;
+    relay: string;
+  }): Promise<boolean> {
+    // Convert the DID document and DID metadata (such as DID types) to a DNS packet.
+    const dnsPacket = await DidDhtDocument.toDnsPacket({
+      didDocument : did.didDocument,
+      didMetadata : did.metadata
+    });
+
+    // Create a signed BEP44 put message from the DNS packet.
+    const bep44Message = await DidDhtUtils.createBep44PutMessage({
+      dnsPacket,
+      publicKeyBytes : DidDhtUtils.identifierToIdentityKeyBytes({ didUri: did.uri }),
+      signer         : await did.getSigner()
+    });
+
+    // Publish the DNS packet to the DHT network.
+    const putResult = await DidDhtDocument.pkarrPut({ relay, bep44Message });
+
+    return putResult;
+  }
+
+  /**
+   * Retrieves a signed BEP44 message from a Pkarr relay server.
+   *
+   * @see {@link https://github.com/Nuhvi/pkarr/blob/main/design/relays.md | Pkarr Relay design}
+   *
+   * @param relay - The Pkarr relay server URL.
+   * @param publicKeyBytes - The public key bytes of the Identity Key, z-base-32 encoded.
+   * @returns A promise resolving to a BEP44 message containing the signed DNS packet.
+  */
+  private static async pkarrGet({ relay, publicKeyBytes }: {
+    publicKeyBytes: Uint8Array;
+    relay: string;
+  }): Promise<Bep44Message> {
+    // The identifier (key in the DHT) is the z-base-32 encoding of the Identity Key.
+    const identifier = Convert.uint8Array(publicKeyBytes).toBase32Z();
+
+    // Concatenate the Pkarr relay URL with the identifier to form the full URL.
+    const url = new URL(identifier, relay).href;
+
+    // Transmit the Get request to the Pkarr relay and get the response.
+    let response: Response;
+    try {
+      response = await fetch(url, { method: 'GET' });
+
+      if (!response.ok) {
+        throw new DidError(DidErrorCode.NotFound, `Pkarr record not found for: ${identifier}`);
+      }
+
+    } catch (error: any) {
+      if (error instanceof DidError) throw error;
+      throw new DidError(DidErrorCode.InternalError, `Failed to fetch Pkarr record: ${error.message}`);
+    }
+
+    // Read the Fetch Response stream into a byte array.
+    const messageBytes = await response.arrayBuffer();
+
+    if (messageBytes.byteLength < 72) {
+      throw new DidError(DidErrorCode.InvalidDidDocumentLength, `Pkarr response must be at least 72 bytes but got: ${messageBytes.byteLength}`);
+    }
+
+    if (messageBytes.byteLength > 1072) {
```

Comment:
do you need >= 1072 ?

"Pkarr response exceeds 1000 byte limit"
I guess its 72 bytes for base did and then 1000 bytes for other ?

Reply:
Yes.  The Mainline DHT max payload is 1000 bytes but `messageBytes` includes a 64 byte signature + 8 byte sequence number